pil-analyzer/tests/types.rs

```diff
@@ -614,3 +730,50 @@ fn new_fixed_column_wrong_type() {
     "#;
     type_check(input, &[]);
 }
+
+#[test]
+fn trait_with_user_defined_enum() {
+    let input = "
+    enum Bool { True, False }
+    
+    trait Not<T> {
+        not: T -> T,
+    }
+    
+    impl<T> Not<Bool> {
+        not: |b| match b {
+            Bool::True => Bool::False,
+            Bool::False => Bool::True,
+        },
+    }
+    let b: Bool = Not::not(Bool::True);
```

Comment:
```suggestion
    let b = Not::not(Bool::True);
```
Does this work?

Reply:
Yes, this one works (probably because there is only one impl). 
The one that doesn't work is the following case, `Convert`.